autonat.go

```diff
@@ -336,6 +374,23 @@ func (as *AmbientAutoNAT) probeNextPeer() {
 		if proto, err := as.host.Peerstore().SupportsProtocols(p, AutoNATProto); len(proto) == 0 || err != nil {
 			continue
 		}
+
+		goodAddr := false
+		for _, a := range info.Addrs {
+			if !as.config.skipDial(a) {
+				goodAddr = true
+				// if a public IP of the peer is one of ours: skip the peer.
+				aIP, _ := manet.ToIP(a)
+				aHost, _ := manet.FromIP(aIP)
```

Comment:
We can't ignore these errors, can we? What if the address doesn't have an IP address, we'll call `manet.FromIP` on `nil`.